daliuge-engine/dlg/apps/simple.py

```diff
@@ -533,7 +533,7 @@ class ArrayGatherApp(BarrierAppDROP):
         [dlg_batch_output("binary/*", [])],
         [dlg_streaming_input("binary/*")],
     )
-    value_list = dlg_list_param("value_list", [])
+    # value_list = dlg_list_param("value_list", [])
```

Comment:
This value is used in the `readWriteData` method; I recognise that the `run` is called before `readWriteData`, but we should either initialise it earlier, or enforce that `readWriteData` can't be called before `run()`.